ev3dev2/motor.py

```diff
@@ -2099,6 +2123,178 @@ def follow_line(self,
+        speed_native_units = speed.to_native_units(self.left_motor)
+        MAX_SPEED = SpeedNativeUnits(self.max_speed)
+
+        while follow_for(self, **kwargs):
+            current_angle = self.__gyro.angle
+            error = current_angle - target_angle
+            integral = integral + error
+            derivative = error - last_error
+            last_error = error
+            turn_native_units = (kp * error) + (ki * integral) + (kd * derivative)
+
+            left_speed = SpeedNativeUnits(speed_native_units - turn_native_units)
+            right_speed = SpeedNativeUnits(speed_native_units + turn_native_units)
+
+            if left_speed > MAX_SPEED:
+                log.info("%s: left_speed %s is greater than MAX_SPEED %s" %
+                        (self, left_speed, MAX_SPEED))
+                self.stop()
+                raise FollowGyroAngleErrorTooFast(
+                    "The robot is moving too fast to follow the angle")
+
+            if right_speed > MAX_SPEED:
+                log.info("%s: right_speed %s is greater than MAX_SPEED %s" %
+                        (self, right_speed, MAX_SPEED))
+                self.stop()
+                raise FollowGyroAngleErrorTooFast(
+                    "The robot is moving too fast to follow the angle")
+
+            if sleep_time:
+                time.sleep(sleep_time)
+
+            self.on(left_speed, right_speed)
+
+        self.stop()
+
+    def turn_to_angle_gyro(self,
+            speed,
+            target_angle=0,
+            sleep_time=0.01
+        ):
+        """
+        Pivot Turn
+
+        ``speed`` is the desired speed of the midpoint of the robot
+
+        ``target_angle`` is the target angle we want to pivot to
+
+        ``sleep_time`` is how many seconds we sleep on each pass through
+            the loop.  This is to give the robot a chance to react
+            to the new motor settings. This should be something small such
+            as 0.01 (10ms).
+
+        Example:
+
+        .. code:: python
+        
+            from ev3dev2.motor import OUTPUT_A, OUTPUT_B, MoveTank, SpeedPercent
+            from ev3dev2.sensor.lego import GyroSensor
+
+            # Instantiate the MoveTank object
+            tank = MoveTank(OUTPUT_A, OUTPUT_B)
+
+            # Initialize the tank's gyro sensor
+            tank.gyro = GyroSensor()
+
+            # Calibrate the gyro to eliminate drift, and to initialize the current angle as 0
+            tank.calibrate_gyro()
+
+            # Pivot 30 degrees
+            tank.turn_to_angle_gyro(
+                speed=SpeedPercent(5),
+                target_angle(30)
+            )
+        """
+        assert self.__gyro, "GyroSensor must be defined"
+
+        speed_native_units = speed.to_native_units(self.left_motor)
+        target_reached = False
+
+        while not target_reached:
+            current_angle = self.__gyro.angle
+            if (abs(current_angle - target_angle) <= 2):
```

Comment:
Would pass the 2 into this method as an arg `wiggle_room=2` and then compare against `wiggle_room` here...that way the user can control how accurate they want it to be.